python/cudf/cudf/core/index.py

```diff
@@ -204,44 +203,36 @@ class RangeIndex(BaseIndex, BinaryOperand):
     def __init__(
         self, start, stop=None, step=1, dtype=None, copy=False, name=None
     ):
-        if step == 0:
-            raise ValueError("Step must not be zero.")
```

Comment:
Was this always an unnecessary restriction? pandas certainly seems to allow decreasing RangeIndex objects when I try this now.

Reply:
I think when `RangeIndex._step` was a parameter this was necessary since `range` doesn't accept an _exactly zero_ step. (This would still allow decreasing step)

Reply:
Ah I misread this as checking `>=0` and thought that didn't seem right and that we must have already supported that. But now I'm confused, why can we remove this check? `step` still needs to be nonzero, right?
```
>>> pd.RangeIndex(10, 20, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/coder/.conda/envs/rapids/lib/python3.10/site-packages/pandas/core/indexes/range.py", line 168, in __new__
    raise ValueError("Step must not be zero")
ValueError: Step must not be zero
```
Are you just relying on the `range` constructor to handle this for you now?
```
>>> range(10, 20, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: range() arg 3 must not be zero
```
If so, should we do a `try-except` to rethrow a more useful exception since the user won't be directly constructing the range?

Reply:
Yeah I was intending to rely on the `range` constructor to validate this, but sure I'll wrap this in a try/except to raise a better message

Reply:
Cool thanks.